### ISSUE TITLE
Dataclass field order flagged when not generating __init__

### ISSUE THREAD

**Issue by Terrance** [microsoft/pyright#1753]
## Describe the bug

Pyright correctly rejects required fields of a dataclass defined after optional ones, but this only applies when generating an `__init__` method -- using `@dataclass(init=False)` to skip its generation allows the class to define fields in any order.

Whilst this can be worked around by reordering the fields of a single class, this becomes more of an issue with subclasses.

## To reproduce

```python
from dataclasses import dataclass

@dataclass(init=False)
class A:
    optional: int = 1
    required: int  # Dataclass fields without default values cannot appear after fields with default values

@dataclass(init=False)
class B(A):
    another: int  # Dataclass fields without default values cannot appear after fields with default values
```

## VS Code extension or command-line

Pyright 1.1.130 in VSCode.

**Comment by erictraut** [comment]
I've re-read PEP 557, and it's not clear on whether the ordering must be enforced regardless of whether an `__init__` method is generated. Mypy enforces the ordering requirement regardless, and I think the authors of PEP 557 were involved in implementing the support within mypy. That seems to indicate that the behavior is intentional. But perhaps this was simply an oversight.

I've posted a [question in the mypy repo](https://github.com/python/mypy/issues/10309) to try to get some advice on this topic.

**Comment by erictraut** [comment]
I've changed the behavior so the field order requirement will not be enforced when `init=False`. This change will appear in the next release of pyright.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.131, which I just published. It was included in the previous version of pylance.
